crates/bevy_ecs/macros/src/component.rs

```diff
@@ -66,48 +65,27 @@ const TABLE: &str = "Table";
 const SPARSE_SET: &str = "SparseSet";
 
 fn parse_component_attr(ast: &DeriveInput) -> Result<Attrs> {
-    let meta_items = bevy_macro_utils::parse_attrs(ast, COMPONENT)?;
-
     let mut attrs = Attrs {
         storage: StorageTy::Table,
     };
 
-    for meta in meta_items {
-        use syn::{
-            Meta::NameValue,
-            NestedMeta::{Lit, Meta},
-        };
-        match meta {
-            Meta(NameValue(m)) if m.path == STORAGE => {
-                attrs.storage = match get_lit_str(STORAGE, &m.lit)?.value().as_str() {
-                    TABLE => StorageTy::Table,
-                    SPARSE_SET => StorageTy::SparseSet,
+    for meta in ast.attrs.iter().filter(|a| a.path().is_ident(COMPONENT)) {
+        meta.parse_nested_meta(|nested| {
+            if nested.path.is_ident(STORAGE) {
+                attrs.storage = match nested.value()?.parse::<LitStr>()?.value() {
+                    s if s == TABLE => StorageTy::Table,
+                    s if s == SPARSE_SET => StorageTy::SparseSet,
```

Comment:
```suggestion
                attrs.storage = match nested.value()?.parse::<LitStr>()?.value().as_str() {
                    TABLE => StorageTy::Table,
                    SPARSE_SET => StorageTy::SparseSet,
```

Reply:
I thought changing to `s if s == FOO` was more clear than `FOO`, because matching on consts is such a dumb feature, and an easy footgun. If `FOO` is not defined, you don't get a compilation error, you bind to the variable `FOO`, turning the match arm into an unconditional capture.

Beyond that, it's fairly difficult to parse if you are not specifically aware of how matching on consts work.

Reply:
> If `FOO` is not defined, you don't get a compilation error, you bind to the variable `FOO`, turning the match arm into an unconditional capture.

Woah, never thought of that 😳

Reply:
I'll keep it like that then 🙂